scipy/spatial/transform/_rigid_transformation.pyx

```diff
@@ -0,0 +1,1979 @@
+        elif n == 1:
+            if self._single:
+                return self.__class__(self._matrix[0], copy=True)
+            else:
+                return self.__class__(self._matrix, copy=True)
+        else:
+            matrix = self.as_matrix()
+            if self._single:
+                matrix = matrix[None, :, :]
+            for i in range(len(matrix)):
+                matrix[i] = np.linalg.matrix_power(matrix[i], abs(n))
+            if self._single:
+                matrix = matrix[0]
+
+            Tf = self.__class__(matrix, copy=False)
+            if n < 0:
+                return Tf.inv()
+            else:
+                return Tf
+
+    @cython.embedsignature(True)
+    def inv(self):
+        """Invert this transformation.
+
+        Composition of a transformation with its inverse results in an identity
+        transformation.
+
+        A rigid transformation is a composition of a rotation and a
+        translation, where the rotation is applied first, followed by the
+        translation. So the inverse transformation is equivalent to the inverse
+        translation followed by the inverse rotation.
+
+        Returns
+        -------
+        `RigidTransformation` instance
+            The inverse of this transformation.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as Tf
+        >>> import numpy as np
+
+        A transformation composed with its inverse results in an identity
+        transformation:
+
+        >>> rng = np.random.default_rng(seed=123)
+        >>> tf = Tf.random(rng=rng)
+        >>> tf.as_matrix()
+        array([[-0.27687894,  0.08111092, -0.95747536,  0.21419513],
+               [ 0.43673235, -0.87694466, -0.20058143,  0.19815685],
+               [-0.85592225, -0.47369724,  0.20738374, -0.95648017],
+               [ 0.        ,  0.        ,  0.        ,  1.        ]])
+
+        >>> (tf.inv() * tf).as_matrix()
+        array([[[1., 0., 0., 0.],
+                [0., 1., 0., 0.],
+                [0., 0., 1., 0.],
+                [0., 0., 0., 1.]]])
+
+        The inverse rigid transformation is the same as the inverse translation
+        followed by the inverse rotation:
+
+        >>> t, r = tf.as_components()
+        >>> r_inv = r.inv()  # inverse rotation
+        >>> t_inv = -t  # inverse translation
+        >>> tf_r_inv = Tf.from_rotation(r_inv)
+        >>> tf_t_inv = Tf.from_translation(t_inv)
+        >>> np.allclose((tf_r_inv * tf_t_inv).as_matrix(),
+        ...             tf.inv().as_matrix(),
+        ...             atol=1e-12)
+        True
+        >>> (tf_r_inv * tf_t_inv * tf).as_matrix()
+        array([[[1., 0., 0., 0.],
+                [0., 1., 0., 0.],
+                [0., 0., 1., 0.],
+                [0., 0., 0., 1.]]])
+        """
+        r_inv = self.rotation.inv()
+        t_inv = -r_inv.apply(self.translation)
+        return self.__class__.from_components(t_inv, r_inv)
```

Comment:
I think this is borderline too inefficient, perhaps implement in terms of the resulting transform matrix.

Reply:
Agreed, updated to build the matrix directly.